### PR TITLE
Add tests for REST/Vector/CRUD edge cases

### LINKED ISSUE #2715: Add tests for fail/edge cases of Vector+REST CRUD operations
(note: follow-up to #2653)

Initial implementation tests happy-case of implementation, but we need to add tests for:

* Invalid TABLE (too big vector column; non-float element type vector column)
* Unusual Vector value (null, missing) that should work
* Invalid Vector value (too big, wrong element type)

to ensure handling works appropriately for these too.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds tests (and related fixes, where necessary) to try out non-happy-path use of Vector CRUD operations (but not vector search) for REST API

**Which issue(s) this PR fixes**:
Fixes #2715

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Ready for review; one of 2 `@Disabled` may be due to our version of `dse-next` backend not validating vector element type (see #2745) -- will try to work on that issue next and see if I can resolve it.

**Comment by tatu-at-datastax** [comment]
Was able to resolve things with newer DSE-next persistence so that only 1 IT `@Disabled` -- one checking there's max vector length. This guard-rail will likely be re-introduced soon into `dse-db-all` but doesn't seem to be available or on by default, yet.

**Review by jeffreyscarpenter — APPROVED** [review]

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2VectorTestIT.java [hunk 1]
```diff
@@ -48,21 +52,62 @@ public void tableCreateWithVectorIndex() {
...
+    assertThat(apiError.code()).isEqualTo(HttpStatus.SC_BAD_REQUEST);
+    assertThat(apiError.description())
+        .contains("INVALID_ARGUMENT")
+        .contains("vectors may only use float. given int");
+  }
+
+  // 24-Aug-2021, tatu: Current version of dse-next does not have guard rails yet,
+  //    cannot yet add test
+  @Test
+  @Disabled("Cannot test 'too big vector' before dse-next backend version supports it")
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2VectorTestIT.java:79]: I did re-test and did not see a failure even after upgrade. I can re-test quickly, first, then ask on vector channel (jd should know) if guard-rails are included yet.

[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/RestApiV2VectorTestIT.java:79]: After upgrading to even later `dse-db-all`, talking to JD, it seems like there may be something else going on.

I added similar tests for CQL-over-REST (/cql) end point and similar behavior is seen. For last thing I'll see if plain CQL tests can be used to further shed light on the issue.
